src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,106 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private Node[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> newNode = new Node<>(key == null ? 0 : key.hashCode(), null, key, value);
+        if (table[getBucket(key)] == null) {
+            if (checkSizeOfTable()) {
+                put(key, value);
+                return;
+            }
+            table[getBucket(key)] = newNode;
+        } else {
+            Node<K, V> currentNode = table[getBucket(key)];
+            Node<K, V> lastNode = currentNode;
+            while (currentNode != null) {
+                if ((currentNode.key == newNode.key)
+                        || (currentNode.key != null && currentNode.key.equals(newNode.key))) {
```

Comment:
from this topic you can use `Objects.equals()`, it will make your code easier to read